src/rgw/services/svc_user_rados.cc

```diff
@@ -738,7 +742,10 @@ int RGWSI_User_RADOS::list_buckets(RGWSI_MetaBackend::Context *ctx,
   int ret;
 
   buckets->clear();
-  
+   if (user.id == RGW_USER_ANON_ID) {
+    ldout(cct, 20) << "RGWSI_User_RADOS::list_buckets(): anonymous user" << dendl;
+    return 0;
```

Comment:
if this is returning success i think it should clear `*is_truncated = false` so the caller doesn't try to loop